misc/mcp_server_stateless.py

```diff
@@ -0,0 +1,102 @@
+# ---
+# cmd: ["modal", "serve", "misc/mcp_server_stateless.py"]
+# ---
+#
+# # Deploy a remote, stateless MCP server on Modal
+#
+# This example demonstrates how to deploy a simple [MCP
+# server](https://modelcontextprotocol.io/) on Modal, that provides a tool to get the
+# current date and time in a given timezone. It is a stateless MCP server, meaning that
+# it does not store any state between requests, and uses the "streamable HTTP" transport
+# type.
+#
+# It uses the [FastMCP library](https://github.com/jlowin/fastmcp) to create the MCP
+# server, which is wrapped using FastAPI, and served with a [Modal web
+# endpoint](https://modal.com/docs/guide/webhooks).
+
+
+import modal
+
+app = modal.App("datetime-mcp-server")
+
+image = modal.Image.debian_slim(python_version="3.12").pip_install(
+    "fastapi==0.115.14",
+    "fastmcp~=2.10.2",
+)
+
+with image.imports():
+    from fastapi import FastAPI
+    from fastmcp import FastMCP
+
+# First, we create the MCP server itself, using FastMCP.
+mcp = FastMCP("Date and Time MCP Server")
+
+
+# Next, we add a tool to the MCP server, that allows LLMs to get the current date and
+# time in a given timezone.
+@mcp.tool()
+async def current_date_and_time(timezone: str = "UTC") -> str:
+    """Get the current date and time.
+
+    Args:
+        timezone: The timezone to get the date and time in (optional). Defaults to UTC.
+
+    Returns:
+        The current date and time in the given timezone, in ISO 8601 format.
+    """
+    from datetime import datetime
+    from zoneinfo import ZoneInfo
+
+    try:
+        tz = ZoneInfo(timezone)
+    except Exception:
+        raise ValueError(
+            f"Invalid timezone '{timezone}'. Please use a valid timezone like 'UTC', "
+            "'America/New_York', or 'Europe/Stockholm'."
+        )
+    return datetime.now(tz).isoformat()
+
+
+# We then use FastMCP to create a Starlette app with `streamable-http` as transport
+# type, and set `stateless_http=True` to make it stateless. This will then be mounted by
+# FastAPI.
+mcp_app = mcp.http_app(transport="streamable-http", stateless_http=True)
+
+fastapi_app = FastAPI(lifespan=mcp_app.router.lifespan_context)
+fastapi_app.mount("/", mcp_app, "mcp")
```

Comment:
This shouldn't be in global scope right? Because then the user needs `fastmcp` and `fastapi` installed locally. It does make writing the doc text harder though.

Reply:
In case it makes the structuring easier: the current rule for examples is that you can have fastapi imports and other objects in the global scope, since we so often need that for the `fastapi_endpoint` feature. Still have to move `fastmcp` and `@mcp` though.

Reply:
Thanks, pushed a change to move them out now!